go/database/mpt/archive_trie.go

```diff
@@ -75,15 +75,33 @@ func OpenArchiveTrie(directory string, config MptConfig, cacheCapacity int) (*Ar
 	}, nil
 }
 
-func VerifyArchive(directory string, config MptConfig, observer VerificationObserver) error {
```

Comment:
I think this method should have been extended with `codes` verification, otherwise untouched. The other method(s) I mention should not have been introduced. It makes the design somehow not easy to follow.